scripts/install-zenml-dev.sh

```diff
@@ -28,7 +28,9 @@ parse_args () {
 
 install_zenml() {
     # install ZenML in editable mode
-    uv pip install $PIP_ARGS -e ".[server,templates,terraform,secrets-aws,secrets-gcp,secrets-azure,secrets-hashicorp,s3fs,gcsfs,adlfs,dev,mlstacks]"
+    uv pip install $PIP_ARGS -e ".[server,templates,terraform,secrets-aws,secrets-gcp,secrets-azure,secrets-hashicorp,s3fs,gcsfs,adlfs,dev]"
+    # TODO: Remove and add the extra back in
+    uv pip install $PIP_ARGS git+https://github.com/zenml-io/mlstacks.git@feature/upgrade-python-3.12
```

Comment:
Please remember to remove this

Reply:
Thank you for the reminder. I had to add one more of this to the `small-checks` workflow in the CI, unfortunately. We have to keep them until we can release a new version of `mlstacks`. I will keep this thread unresolved as a reminder to myself.

Reply:
@stefannica Just so you know, this is going to be a bit tricky as I can not remove these lines with this PR. 

Right now, the changes that I made are not compatible with the latest release of `mlstacks`(0.9.0). If I remove them now, all the subsequent CI runs on other PRs will fail. 

We can potentially release a new version of `mlstacks` as a solution but if we do that `mlstacks` will be broken until we release a new version of `zenml`. (There won't be any `zenml` versions which can work with this new `mlstacks` release.)

So, my proposal would be something like this:

1. I merge this PR while keeping `install-zenml-dev` and `small-checks` pointing to my upgrade branch on the `mlstacks` repo.
2. I create a follow-up PR which actually reverts them back.
3. Just before we release a new version of `zenml`, we do the `mlstacks` release, merge the PR from point 2 and proceed with the release.

This needs to be communicated to the product team as well, I will take over that.